app/client/src/actions/pageActions.tsx

```
@@ -220,6 +220,12 @@ export const clonePageSuccess = (
 };
 
 export const updatePage = (payload: UpdatePageRequest) => {
+  // Update page *needs* id to be there. We found certain scenarios
+  // where this was not happening and capturing the error to know gather
+  // more info: https://github.com/appsmithorg/appsmith/issues/16435
+  if (!payload.id) {
+    throw new Error("Attempting to update page without page id");
```

Curious to understand which error page would be shown for this failure?  @hetunandu

---

I thought it would be the generic error page but it seemed to fail silently with just logging. Updated to log on sentry instead